### PR TITLE
Html css popup

### PR DESCRIPTION
**Pull request by vl-shevchenko**
### html-css-popup

[Demo](https://vl-shevchenko.github.io/popup/) |
[Code base](https://github.com/vl-shevchenko/popup)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by vl-shevchenko** [comment]
Hey!
My code has been validated.
I took into account all the tips from common mistakes when writing code.

File: submissions/vl-shevchenko/html-css-popup/style.css [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    margin-right: 50px;
+    
+}
+
+#checkbox_knopka {
+    appearance: none;
+    position: absolute;
+}
+
+.list a{
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/style.css:62]: Avoid styling by HTML tags except if you add base styles. [Reason](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css). Also, avoid too specific CSS selectors. [Read about it](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)

File: submissions/vl-shevchenko/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/news.png" alt="">News</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/maps.png" alt="">Map</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/files.png" alt="">Files</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/contacts.png" alt="">Contacts</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/videos.png" alt="">Video</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/apps.png" alt="">Apps</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/promo.png" alt="">Promo</a></li>
+                                        </ul>
+                                        <input tabindex="-1" type="checkbox" id="checkbox_more">
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/index.html:36]: `tabindex="-1"` don't allow to get browser focus and this button is not accessible from the keyboard.

File: submissions/vl-shevchenko/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                <div>                                 
+                    <div class="conteiner_menu">                   
+                        <ul class="conteiner__unlist">
+                            <li class="list"><a href="#">Mail</a></li>
+                            <li class="list"><a href="#">Images</a></li>
+                            <li>
+                                <input tabindex="-1" class="checkbox_knopka" type="checkbox" id="checkbox_knopka">
+                                <label class="list" for="checkbox_knopka">                               
+                                    <a tabindex="0"><img src="popup-icons2/01.png" alt=""></a>
+                                </label>
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/index.html:23]: `tabindex="-1"` don't allow to get browser focus and this button is not accessible from the keyboard. Also, `tabindex="0"` makes no sense on interactive elements.

File: submissions/vl-shevchenko/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/contacts.png" alt="">Contacts</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/videos.png" alt="">Video</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/apps.png" alt="">Apps</a></li>
+                                            <li class="list-hiden"><a href="#"><img src="popup-icons2/promo.png" alt="">Promo</a></li>
+                                        </ul>
+                                        <input tabindex="-1" type="checkbox" id="checkbox_more">
+                                        <label tabindex="0" class="checkbox_more-btn more-btn" for="checkbox_more" >
+                                            <a class="more-btn">More</a>
+                                        </label>                         
+                                        <hr class="more-hr">
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/index.html:40]: Avoid using HTML elements for decorative purposes use CSS for it(pseudoelements, borders, etc)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Awesome design!

**Comment by vl-shevchenko** [comment]
Hey!
Bug fix - done.

**Comment by A-Ostrovnyy** [comment]
For the future homework when you pushed the changes and they ready for review click `re-request review` button
image

File: submissions/vl-shevchenko/html-css-popup/index.html [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                            <li class="list"><a class="list-link" href="#">Mail</a></li>
+                            <li class="list"><a class="list-link" href="#">Images</a></li>
+                            <li>
+                                <input class="checkbox_knopka" type="checkbox" id="checkbox_knopka">
+                                <label class="list" for="checkbox_knopka">                               
+                                    <a class="list-link"><img src="popup-icons2/01.png" alt=""></a>
+                                <input tabindex="-1" class="checkbox_knopka" type="checkbox" id="checkbox_knopka">
+                                <label class="list" for="checkbox_knopka">                               
+                                    <a tabindex="0"><img src="popup-icons2/01.png" alt=""></a>
+                                </label>
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/index.html:26]: What about these `tabindex` attributes, be sure you fix all elements with `tabindex` attribute

File: submissions/vl-shevchenko/html-css-popup/style.css [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+.conteiner__unlist {
+    display: flex;
+    align-items: center;
+    text-align: center;
+    margin-right: 50px;
+    
+}
+
+#checkbox_knopka {
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/style.css:63]: Avoid transliteration in naming entities.

File: submissions/vl-shevchenko/html-css-popup/style.css [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+    padding: 0px 15px;
+    height: 55px;
+    font-size: 20px;
+    text-align: center;
+    color: #242220;
+    border-radius: 10px;
+    cursor: pointer;
+}
+
+.list img {
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/style.css:80]: Same story: avoid styling by HTML tags

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good progress!

**Comment by vl-shevchenko** [comment]
I am fix naming and delete styling by HTML tags.

File: submissions/vl-shevchenko/html-css-popup/style.css [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+.list-link:focus  {
+    background: rgba(241, 241, 241, 0.74);
+    border-radius: 10px;
+    border-bottom: 3px solid rgb(61, 68, 71);
+    
+}
+
+
+#checkbox_btn:focus ~ label > a {
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/style.css:101]: Avoid styling by HTML tags except if you add base styles. [Reason](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css). Avoid using too specific CSS selectors. [Read more about that](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity).

**Comment by vl-shevchenko** [comment]
But I see "tabindex"... File index.html in my repo not have "tabindex". 
Why does this happen?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
One last issue)

**Comment by A-Ostrovnyy** [comment]
> But I see "tabindex"... File index.html in my repo not have "tabindex". Why does this happen?

In what file/repository do you see `tabindex`?

**Comment by vl-shevchenko** [comment]
> 
 [I see in this rerepository](https://github.com/kottans/frontend-2022-homeworks). in file index.html

 [Screenshot](https://github.com/vl-shevchenko/popup/blob/main/Screenshot%202022-08-27%20221921.jpg)

**Comment by A-Ostrovnyy** [comment]
The code with red background is a deleted code. To see the final version of your code look to the `Files changed` tab
image

**Comment by vl-shevchenko** [comment]
Sorry please. Thanks for the explanation.

File: submissions/vl-shevchenko/html-css-popup/style.css [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    color: #242220;
+    border-radius: 10px;
+    cursor: pointer;
+    justify-content: space-around;
+}
+
+
+
+
+
```
Comment (A-Ostrovnyy) [inline submissions/vl-shevchenko/html-css-popup/style.css:83]: Avoid too many unused lines in your work.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
See all unresolved comments
Good progress!

**Comment by vl-shevchenko** [comment]
Hey!
I deleted styling by HTML tags and unused lines.

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**